pkg/kubelet/dockershim/network/cni/cni_others.go

```diff
@@ -60,6 +61,11 @@ func (plugin *cniNetworkPlugin) platformInit() error {
 // TODO: Use the addToNetwork function to obtain the IP of the Pod. That will assume idempotent ADD call to the plugin.
 // Also fix the runtime's call to Status function to be done only in the case that the IP is lost, no need to do periodic calls
 func (plugin *cniNetworkPlugin) GetPodNetworkStatus(namespace string, name string, id kubecontainer.ContainerID) (*network.PodNetworkStatus, error) {
+
```

Comment:
Should have a comment here about the ordering of how to get the pod's IP address.

But really, we should simply stop jumping into the namespace to check the pod's IP, and just rely on the CNI plugin's result since that result is authoritative.  Then we can get rid of the code in docker_sandbox.go in getIP() that checks Docker's NetworkSettings too.  Some of the Windows code could also be consolidated perhaps.

Reply:
I didn't remove code related to nsenter, because I hope to keep compatibility with CNI plugins which relies on it. But frankly saying I don't know such plugins, I see workarounds in kuryr-kubernetes and intel's vhostuser can work only in couple with multus - it's the same workaround.
So lets  remove nsenter code ;)
Do you expect that in separate patch?

Reply:
I found, just removing nsenter approach it's not good idea, due to GetPodNetworkStatus is called after kubelet restart, and IP in plugin is empty. I think it's better to store that ip in etcd for that case.